packages/material-ui/src/Modal/TrapFocus.js

```diff
@@ -100,7 +100,13 @@ function TrapFocus(props) {
     doc.addEventListener('focus', contain, true);
     doc.addEventListener('keydown', loopFocus, true);
 
+    const interval = setInterval(() => {
+      contain();
+    }, 500);
```

Comment:
500ms is really long. Shouldn't this be 50?

Reply:
I don't know. The value should be large enough not to put much pressure on the main thread but be small enough not to disturb user interactions. The average reaction time for a human is around 700ms, e.g. for a car accident. Maybe the value could be defined with a heuristic: pick the reproduction example, start from 50ms, increase the value up until you can no longer make the escape keypress, consequent to an enter keypress work?

Reply:
>100ms is the magic number when it comes to noticing time passed. If 400ms puts already pressure on the main thread your app has other issues.

`contain` is currently expensive? Maybe just check if the focus is on the body (which is the case if it is lost like in the reported issue) and only then run it. This should allow us to reduce it to 50ms which is ~3 frames. That seems alot less likely to be noticeable.

Reply:
Yeah, sure, that makes sense to me. No matter what we pick, the cost of changing it is low. Can it have an impact of people perception if the see something that ticks every 50ms when debugging a perf issue on their app? Or, can it have an impact on the requestidlecallback API? I have no idea. The future will tell us 🤷‍♂️. +1 for trying 50ms.